regparser/tree/xml_parser/reg_text.py

```diff
@@ -224,8 +224,11 @@ def build_from_section(reg_part, section_xml):
 
     nodes = []
     section_nums = []
-    for match in re.finditer(r'%s\.(\d+)' % reg_part, section_no):
-        section_nums.append(int(match.group(1)))
+    for match in re.finditer(r'%s\.(\d+[:alpha:]*)' % reg_part, section_no):
+        secnum_candidate = match.group(1)
+        if all([char.isdigit() for char in secnum_candidate]):
```

Comment:
I think you can call `isdigit()` on the whole string, hey? Also, no need to create a list since you are throwing away the sequence. Does this work?
```python
if secnum_candidate.isdigit():
    secnum_candidate = int(secnum_candidate)
```